airflow/providers/google/cloud/triggers/gcs.py

```diff
@@ -97,3 +99,101 @@ async def _object_exists(self, hook: GCSAsyncHook, bucket_name: str, object_name
             if object_response:
                 return "success"
             return "pending"
+
+
+class GCSCheckBlobUpdateTimeTrigger(BaseTrigger):
+    """
+    A trigger that makes an async call to GCS to check whether the object is updated in a bucket.
+
+    :param bucket: google cloud storage bucket name cloud storage where the objects are residing.
+    :param object_name: the file or folder present in the bucket
+    :param ts: datetime object
+    :param poke_interval: polling period in seconds to check for file/folder
+    :param google_cloud_conn_id: reference to the Google Connection
+    :param hook_params: dict object that has delegate_to and impersonation_chain
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        object_name: str,
+        ts: datetime,
+        poke_interval: float,
+        google_cloud_conn_id: str,
+        hook_params: dict[str, Any],
+    ):
+        super().__init__()
+        self.bucket = bucket
+        self.object_name = object_name
+        self.ts = ts
+        self.poke_interval = poke_interval
+        self.google_cloud_conn_id: str = google_cloud_conn_id
+        self.hook_params = hook_params
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        """Serializes GCSCheckBlobUpdateTimeTrigger arguments and classpath."""
+        return (
+            "airflow.providers.google.cloud.triggers.gcs.GCSCheckBlobUpdateTimeTrigger",
+            {
+                "bucket": self.bucket,
+                "object_name": self.object_name,
+                "ts": self.ts,
+                "poke_interval": self.poke_interval,
+                "google_cloud_conn_id": self.google_cloud_conn_id,
+                "hook_params": self.hook_params,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:
+        """Simple loop until the object updated time is greater than ts datetime in bucket."""
```

Comment:
```suggestion
        """Loop until the object updated time is greater than ts datetime in bucket."""
```

we don't nec. need to assert that this loop is simple :) we can let the reader be the judge of that 😛

Reply:
corrected